app/common/lib/bookmarkUtil.js

```diff
@@ -260,18 +154,87 @@ const getKey = (siteDetail) => {
   return null
 }
 
+const buildBookmark = (state, bookmarkDetail) => {
+  bookmarkDetail = makeImmutable(bookmarkDetail)
+  let location
+  if (bookmarkDetail.has('location')) {
+    location = UrlUtil.getLocationIfPDF(bookmarkDetail.get('location'))
+    bookmarkDetail = bookmarkDetail.set('location', location)
+  }
+
+  const key = getKey(bookmarkDetail)
+  const historyKey = key.slice(0, -2)
+  let dataItem = historyState.getSite(state, historyKey)
+
+  if (dataItem.isEmpty()) {
+    // check if we have data in tabs
+    const tab = tabState.getActiveTab(state) || Immutable.Map()
+    const activeLocation = tab.get('url') || tab.getIn(['frame', 'location'])
+
+    if (!tab.isEmpty() && bookmarkDetail.get('location') === activeLocation) {
+      dataItem = makeImmutable({
+        partitionNumber: tab.getIn(['frame', 'partitionNumber'], 0),
+        favicon: tab.getIn(['frame', 'icon']),
+        themeColor: tab.getIn(['frame', 'themeColor'])
+      })
+    } else {
+      // check if bookmark is in top sites
+      const topSites = Immutable.fromJS(newTabData.topSites.concat(newTabData.pinnedTopSites))
+      const topSite = topSites.find(site => site.get('location') === bookmarkDetail.get('location')) || Immutable.Map()
+
+      if (!topSite.isEmpty()) {
+        dataItem = topSite
+      }
+    }
+  }
+
+  return makeImmutable({
+    title: bookmarkDetail.get('title', ''),
+    location: bookmarkDetail.get('location'),
+    parentFolderId: ~~bookmarkDetail.get('parentFolderId', 0),
```

Comment:
Noticed that below you used Number instead of `~~` here. Maybe use `~~` as well here? It's a bit diff in that `~~` is like `Math.floor`. But I think this is always an integer.

Reply:
there was a discussion about `~~` vs `Number` here https://github.com/brave/browser-laptop/pull/10325#pullrequestreview-57656377 and because it was not readable I changed it to Number in some parts. Other `~~` are fixed here https://github.com/brave/browser-laptop/pull/10610/files

Reply:
I don't care which way it was just odd to me to see `~~` being changed to `Number` in some places and then also new `~~` being used.

Reply:
yeah after this PR and #10610 are merged there will be no more `~~`